proto/rill/local/v1/api.proto

```diff
@@ -42,3 +51,56 @@ message GetVersionResponse {
   string current = 1;
   string latest = 2;
 }
+
+message DeployValidationRequest {}
+
+message DeployValidationResponse {
+  bool is_authenticated = 1; // if true below fields are relevant after login
+  bool is_github_connected = 2; // if true below fields are relevant after git install
+  string git_grant_access_url = 3; // redirect to this if is_github_connected or is_repo_access_granted is false
+  string git_user_name = 4;
+  bool is_user_app_installed = 5; // if rill git app is installed on user account
+  string user_app_permission = 6;
+  repeated string git_user_orgs = 7;
+  bool is_git_repo = 8;
+  bool git_remote_found = 9; // only applicable when is_git_repo is true
+  bool is_repo_access_granted = 10; // relevant only when is_git_repo is true and remote found, if false redirect to git_grant_access_url
+  string git_url = 11; // only applicable when is_git_repo is true and remote found
+  UncommittedChanges uncommitted_changes = 12; // only applicable when is_git_repo is true and remote found
+  bool rill_org_exists_as_git_user_name = 13; // only applicable when user does not have any orgs
+  repeated string rill_user_orgs = 14;
+  string local_project_name = 15;
+  repeated OrganizationWithApp organizations_with_app = 16;
+}
+
+enum UncommittedChanges {
+  UNCOMMITTED_CHANGES_UNSPECIFIED = 0;
+  UNCOMMITTED_CHANGES_YES = 1;
+  UNCOMMITTED_CHANGES_NO = 2;
+}
+
+message OrganizationWithApp {
+  string org = 1;
+  string permission = 2;
+}
+
+message PushToGitRequest {
+  string org = 1;
+  string repo = 2;
+}
+
+message PushToGitResponse {
+  string git_url = 1;
```

Comment:
Is this a Git URL (ends in `.git`) or a Github URL (for opening in a browser)? If the second, change to `github_`